### PR TITLE
Refactor `update...()` methods

### PR DESCRIPTION
**Pull request by Tigrov**
* Fix bug: `ActiveRecord::save()` without changes should return `true`;
* Fix bug with return type of `ActiveRecord::getOldPrimaryKey()` method;
* Remove redundant `false` return type in `ActiveRecordInterface::update()` method;
* Refactor `BaseActiveRecord::updateAttributes()`;
* Refactor `BaseActiveRecord::updateInternal()`;
* Refactor `BaseActiveRecord::updateCounters()`.

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/BaseActiveRecord.php [hunk 1]
```diff
@@ -653,7 +644,9 @@ public function save(array $attributeNames = null): bool
             return $this->insert($attributeNames);
         }
 
-        return $this->update($attributeNames) !== false;
+        $this->update($attributeNames);
+
+        return true;
```
Comment (darkdef) [inline src/BaseActiveRecord.php:649]: Always true? And if changed 0 rows?

Comment (Tigrov) [reply src/BaseActiveRecord.php:649]: If `0` rows changed then there were no changes. Operation was successfully completed.

```php
$customerQuery = new ActiveQuery(Customer::class, $this->db);
$customer = $customerQuery->findOne(1);
$customer->save(); // should be true
```

File: src/ActiveRecord.php [hunk 2]
```diff
@@ -366,10 +366,6 @@ protected function deleteInternal(): false|int
         $condition = $this->getOldPrimaryKey(true);
         $lock = $this->optimisticLock();
 
-        if (is_array($condition) === false) {
```
Comment (darkdef) [inline src/ActiveRecord.php:369]: May be add Exception for this behavior?

Comment (Tigrov) [reply src/ActiveRecord.php:369]: `$this->getOldPrimaryKey(true)` returns array always. No need to check `if (is_array($condition)`

**Review by darkdef — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

File: tests/ActiveRecordTest.php [hunk 3]
```diff
@@ -836,4 +836,39 @@ public function testToArrayForArrayable(): void
...
+
+    public function testGetOldPrimaryKey(): void
+    {
+        $this->checkFixture($this->db, 'customer');
+
+        $customerQuery = new ActiveQuery(Customer::class, $this->db);
+
+        $customer = $customerQuery->findOne(1);
+
+        $this->assertSame(1, $customer->getOldPrimaryKey());
```
Comment (xepozz) [inline tests/ActiveRecordTest.php:871]: Let's try to change it before the checks?

Comment (Tigrov) [reply tests/ActiveRecordTest.php:872]: Done

**Review by xepozz — APPROVED** [review]

**Pull request merged**